eurito_daps/routines/example/openaire_simple_pipeline.py

```diff
@@ -0,0 +1,138 @@
+        production (bool): test mode or production mode
+    '''
+    date = luigi.DateParameter(default=datetime.datetime.today())
+    outputType = luigi.Parameter()
+    production = luigi.BoolParameter(default=False)
+
+    def requires(self):
+        '''Call the task to run before this in the pipeline.'''
+
+        logging.getLogger().setLevel(logging.INFO)
+        return CollectOpenAireTask(date=self.date,
+                          outputType=self.outputType,
+                          test=not self.production)
+
+
+class CollectOpenAireTask(luigi.Task):
+    '''Collects OpenAIRE projects and writes them into MySQL database
+
+    Args:
+        date (datetime): Date used to label the outputs
+        outputType (str): type of record to be extracted from OpenAIRE API. Accepts "software", "datasets", "publications", "ECProjects"
+        test (bool): run a shorter version of the task if in test mode
+    '''
+    date = luigi.DateParameter(default=datetime.datetime.today())
+    outputType = luigi.Parameter()
+    test = luigi.BoolParameter()
+
+    def output(self):
+        '''Points to the output database engine where the task is marked as done.
+        The luigi_table_updates table exists in test and production databases.
+        '''
+        db_config = get_config(os.environ["MYSQLDB"], "mysqldb")
+        db_config["database"] = 'dev' if self.test else 'production'
+        db_config["table"] = "Example <dummy>"  # Note, not a real table
+        update_id = "SimpleTask_{}".format(self.date)
+        return MySqlTarget(update_id=update_id, **db_config)
+
+    def run(self):
+        '''Collects records from OpenAIRE API and stores them into the 'dev' or 'production' database
+        '''
+        baseUrl = 'http://api.openaire.eu/oai_pmh?verb=ListRecords'
+
+        #engine = create_engine('sqlite:///pythonsqlite.db', echo=False)
+
+        db = 'dev' if self.test else 'production'
+        engine = get_mysql_engine("MYSQLDBCONF", 'mysqldb', db)
+
+        Base.metadata.create_all(engine)
+
+        #open DB session
+        Session = sessionmaker(bind=engine)
+        globals.dbsession = Session()
+
+        #open up requests session
+        reqsession = requests.session()
+        reqsession.keep_alive = False
+
+        resumptionToken = 'Not None'
+
+        currentUrl = baseUrl + '&metadataPrefix=oaf&set=' + self.outputType
+        logging.info("Current URL:" + currentUrl)
+
+        count = 0
+
+        while resumptionToken != 'None':
+
+            logging.info("Retrieving soup from URL...")
+            cur_soup = openaire_utils.get_soup_contents(currentUrl, reqsession)
+            logging.info("Soup returned...")
+            logging.info(cur_soup)
+
+            titletag = cur_soup.find('title')
+            logging.info(titletag)
+
+            if titletag.text == '503 Service Unavailable' or titletag.text == '502 Proxy Error':

```

Comment:
Pick `http` errors up directly from `requests` using `r.status_code` or `r.raise_for_status()` and catching